src/phylum/init/cli.py

```diff
@@ -111,16 +149,16 @@ def get_archive_url(version, archive_name):
     return archive_url
 
 
-def is_token_set(token=None):
-    """Check if any token is already set in the CLI configuration file.
+def is_token_set(phylum_settings_path, token=None):
+    """Check if any token is already set in the given CLI configuration file.
 
     Optionally, check if a specific given `token` is set.
     """
-    if not SETTINGS_YAML_PATH.exists():
+    if not phylum_settings_path.exists():
         return False
```

Comment:
A more [EAFP](https://docs.python.org/3/glossary.html#term-EAFP) approach would be to just do the `read_text()` call here and see if it fails:

```
    try:
        data = phylum_settings_path.read_text(encoding="utf-8")
    except FileNotFoundError:
        return False
```

(Not using a suggestion here because you would also need to update the line below to use this `data` variable)

Reply:
changed to support suggestion.